registry/storage/driver/s3-aws/s3_test.go

```diff
@@ -623,6 +624,38 @@ func TestWalk(t *testing.T) {
 			},
 			err: false,
 		},
+		{
+			name: "start late without from",
+			fn:   func(fileInfo storagedriver.FileInfo) error { return nil },
+			options: []func(*storagedriver.WalkOptions){
+				storagedriver.WithStartAfterHint("/folder3/subfolder1/subfolder1/file1"),
```

Comment:
Forgive me if I’m missing something obvious here since I’m looking at this on my phone, but why is this `/folder3/subfolder1/…` and not `/folder3/subfolder2/…`?

Reply:
The structure in S3 is this:
```
		"/file1",
		"/file1-2",
		"/file1/2",
		"/folder1/file1",
		"/folder2/file1",
		"/folder3/file1",
		"/folder3/subfolder1/subfolder1/file1",
		"/folder3/subfolder2/subfolder1/file1",
		"/folder4/file1",
		"/folder1-v2/file1",
		"/folder1-v2/subfolder1/file1",
```

My guess is you're thinking it should be 3->2->1, when its folder 3 which has 2 subfolders in it.

Reply:
Thanks for the explanation, that makes sense then.